code/game/gamemodes/miniantags/umbra/umbra.dm

```diff
@@ -0,0 +1,247 @@
+		else if(harvesting)
+			src << "<span class='warning'>You're already trying to harvest a soul!</span>"
+			return
+		harvest_soul(L)
+
+/mob/living/simple_animal/umbra/death()
+	if(breaking_apart)
+		return
+	..(1)
+	breaking_apart = TRUE
+	notransform = TRUE
+	invisibility = FALSE
+	visible_message("<span class='warning'>An [name] appears from nowhere and begins to disintegrate!</span>", \
+	"<span class='userdanger'>You feel your will faltering, and your form begins to break apart!</span>")
+	flick("umbra_disintegrate", src)
+	sleep(12)
+	if(vitae)
+		visible_message("<span class='warning'>[src] breaks apart into a pile of ashes!</span>")
+		var/obj/item/phantasmal_ashes/P = new(get_turf(src))
+		P.umbra_key = key
+		P.umbra_vitae = vitae
+	else
+		visible_message("<span class='warning'>[src] breaks apart and fades away!</span>")
+	qdel(src)
+
+/mob/living/simple_animal/umbra/say() //Umbras can't directly speak
+	return
+
+/mob/living/simple_animal/umbra/attack_ghost(mob/dead/observer/O)
+	if(key)
+		return
+	if(alert(O, "Become an umbra? You won't be clonable!",,"Yes", "No") == "No" || !O)
+		return
+	notify_ghosts("The umbra at [get_area(src)] has been taken control of.", source = src, action = NOTIFY_ORBIT)
+	key = O.key
+	src << playstyle_string
+
+
+
+/mob/living/simple_animal/umbra/proc/adjust_vitae(amount, silent, source)
+	vitae = min(max(0, vitae + amount), vitae_cap)
+	if(!silent)
+		src << "<span class='umbra'>[amount > 0 ? "Gained" : "Lost"] [amount] vitae[source ? " from [source]" : ""].</span>"
+	return vitae
+
+
+
+/mob/living/simple_animal/umbra/proc/harvest_soul(mob/living/L) //How umbras drain vitae from their targets
+	if(!L || !L.stat || L in recently_drained)
+		return
+	harvesting = TRUE
+	src << "<span class='umbra'>You search for [L]'s soul...</span>"
+	if(!do_after(src, 30, target = L))
+		harvesting = FALSE
+		return
+	if(L.hellbound)
+		src << "<span class='warning'>[L] has only emptiness in place of their soul!</span>"
+		harvesting = FALSE
+		return
+	if(!L.stat)
+		src << "<span class='warning'>[L] is conscious again and shields your efforts!</span>"
+		L << "<span class='warning'>You're being watched.</span>"
+		harvesting = FALSE
+		return
+	var/vitae_yield = 1 //A bit of essence even if it's a weak soul
+	var/vitae_information = "<span class='umbra'>This soul is "
+	if(ishuman(L))
+		vitae_information += "human, "
+		vitae_yield += rand(10, 15)
+	if(L.mind)
+		if(!(L.mind in ticker.mode.devils))
+			vitae_information += "sentient, "
+		else
+			vitae_information += "infernal and home to multiple souls, all of which you can drain! Moving on... it's "
+			for(var/i in 1 to (L.mind.devilinfo.soulsOwned.len))
+				vitae_yield += rand(5, 10) //You can drain all the souls that a devil has stolen!
+		vitae_yield += rand(10, 15)
+	if(L.stat == UNCONSCIOUS)
+		vitae_information += "blazing with vitality, "
+		vitae_yield += rand(20, 25) //Significant bonus if the target is in critical condition instead of dead
```

Comment:
This comment is kinda weird, because you can easily be unconscious without being in crit